edb/pgsql/resolver/relation.py

```diff
@@ -479,3 +493,112 @@ def _construct_system_columns() -> List[context.Column]:
     ]
+
+
+def _compile_read_of_obj_table(
+    obj: s_objtypes.ObjectType, include_inherited: bool, ctx: Context
+) -> pgast.Relation:
+    from edb.edgeql import ast as qlast
+    from edb.edgeql import compiler as qlcompiler
+    from edb.pgsql import compiler as pgcompiler
+
+    obj_name: sn.QualName = obj.get_name(ctx.schema)
+    assert obj_name
+    ql_shape = []
+    for ptr in obj.get_pointers(ctx.schema).objects(ctx.schema):
+        assert isinstance(ptr, s_pointers.Pointer)
+        if ptr.get_cardinality(ctx.schema).is_multi():
+            continue
+        ptr_name = ptr.get_shortname(ctx.schema).name
+        ql_shape.append(
+            qlast.ShapeElement(
+                expr=qlast.Path(steps=[qlast.Ptr(name=ptr_name)]),
+            )
+        )
+    ql_stmt = qlast.SelectQuery(
+        result=qlast.Shape(
+            expr=qlast.Path(
+                steps=[
+                    qlast.ObjectRef(module=obj_name.module, name=obj_name.name)
+                ]
+            ),
+            elements=ql_shape,
+        )
+    )
+
+    if not include_inherited:
+        ql_stmt.where = qlast.BinOp(
+            left=qlast.Path(
+                partial=True,
+                steps=[qlast.Ptr(name='__type__'), qlast.Ptr(name='id')],
+            ),
+            op='=',
+            right=qlast.TypeCast(
+                expr=qlast.Constant.string(str(obj.id)),
+                type=qlast.TypeName(maintype=qlast.ObjectRef(name='uuid')),
+            ),
+        )
+
+    ir_stmt = qlcompiler.compile_ast_to_ir(
+        ql_stmt,
+        ctx.schema,
+        options=qlcompiler.CompilerOptions(apply_user_access_policies=True),
+    )
+
+    sql_tree = pgcompiler.compile_ir_to_sql_tree(
+        ir_stmt,
+        output_format=pgcompiler.OutputFormat.NATIVE_INTERNAL,
+        alias_generator=ctx.alias_generator,
+    )
+    command.merge_params(sql_tree, ir_stmt, ctx)
+
+    # add CTEs to resolver's CTE buffer
+    assert isinstance(sql_tree.ast, pgast.Query)
+    if sql_tree.ast.ctes:
+        ctx.ctes_buffer.extend(sql_tree.ast.ctes)
+        sql_tree.ast.ctes.clear()
+
+    # prepare a map from pointer name into pgast
+    ptr_map: Dict[Tuple[str, str], pgast.BaseExpr] = {}
+    for (ptr_id, aspect), pg_expr in sql_tree.ast.path_namespace.items():
+        qual_name = ptr_id.rptr_name()
+        if not qual_name:
+            continue
+        ptr_map[qual_name.name, aspect] = pg_expr
+
+    # collect results and construct the resulting relation with the same
+    # columns as the actual base table
+    sql_tree.ast.target_list.clear()
+    for ptr in obj.get_pointers(ctx.schema).objects(ctx.schema):
+        assert isinstance(ptr, s_pointers.Pointer)
+        ptr_name = ptr.get_shortname(ctx.schema).name
+        if ptr.get_cardinality(ctx.schema).is_multi():
+            continue
+
+        val = ptr_map.get((ptr_name, 'value'), None)
+        if not val:
+            val = ptr_map.get((ptr_name, 'serialized'), None)
```

Comment:
When might it be `serialized`? That makes me a little nervous, since those could be a different tyep sometimes

Reply:
Serialized is actually used the most. This schema type:
```
type Document {
    property name: str;
    link owner: User;
}
```
... gets this path_namespace:

```
((default::Document), <PathAspect.VALUE: 'value'>): <pg.ColumnRef name='q~1.expr~5_value~2' at 0x7f23ad508740>
((default::Document).>id[IS std::uuid], <PathAspect.VALUE: 'value'>): <pg.ColumnRef name='q~1.expr~5_value~2' at 0x7f23ad3c4050>
((default::Document).>__type__[IS schema::ObjectType], <PathAspect.SERIALIZED: 'serialized'>): <pg.SelectStmt at 0x7f23ad508080>
((default::Document).>id[IS std::uuid], <PathAspect.SERIALIZED: 'serialized'>): <pg.ColumnRef name='q~1.expr~5_value~2' at 0x7f23ad3c4050>
((default::Document).>name[IS std::str], <PathAspect.SERIALIZED: 'serialized'>): <pg.SelectStmt at 0x7f23ad6b6660>
((default::Document).>owner[IS default::User], <PathAspect.SERIALIZED: 'serialized'>): <pg.SelectStmt at 0x7f23ad52ef00>
((default::Document), <PathAspect.SERIALIZED: 'serialized'>): <pg.ImplicitRowExpr at 0x7f23ad52f2f0>
```

So, this is what pointers end up using:
- `id`: value
- `__type__`: serialized
- `name`: serialized
- `owner`: serialized